controllers/notificationpolicy_controller.go

```diff
@@ -135,12 +157,104 @@ func (r *GrafanaNotificationPolicyReconciler) Reconcile(ctx context.Context, req
 		return ctrl.Result{}, fmt.Errorf("failed to apply to all instances: %v", applyErrors)
 	}
 
+	if len(mergedRoutes) > 0 {
+		status := statusDiscoveredRoutes(mergedRoutes)
+		notificationPolicy.Status.DiscoveredRoutes = &status
+	}
+
+	if err := r.updateNotificationPolicyRoutesStatus(ctx, notificationPolicy, mergedRoutes); err != nil {
+		r.Log.Error(err, "failed to add merged events to routes")
+	}
+
+	if notificationPolicy.Spec.Route.IsRouteSelectorMutuallyExclusive() {
+		meta.RemoveStatusCondition(&notificationPolicy.Status.Conditions, conditionRoutesIgnoredDueToRouteSelector)
+	} else {
+		setRoutesIgnoredDueToRouteSelectorCondition(&notificationPolicy.Status.Conditions, notificationPolicy.Generation)
+	}
```

Comment:
Is there anything preventing this check from occurring before applying the notification policy? I'd like to avoid half-applied notification policies

Reply:
Ah I see, I expected an error if both are specified but understand what you're going with here. I'm good with this as well!